src/Controller/NewsController.php

```diff
@@ -32,20 +37,55 @@ public function show(?News $news): Response
 
 
     #[Route('/admin/news', name: 'admin_news')]
-    public function adminNews(NewsRepository $newsRepository): Response
+    public function adminNews(NewsRepository $newsRepository)
     {
         return $this->render('news/admin/index.html.twig', [
             'news' => $newsRepository->findBy(array(), array('id' => 'DESC'), 15)
         ]);
     }
 
     #[Route('/admin/create', name: 'admin_news_created')]
-    public function createNews(): Response
+    public function createNews(Request $request, SluggerInterface $slugger, Security $security, EntityManagerInterface $entityManager, WebhookDiscordService $webhookDiscordService)
     {
+        $this->webhookDiscordService = $webhookDiscordService;
         $news = new News();
-        $newForm = $this->createForm(NewsType::class, $news);
+
+        $form = $this->createForm(NewsType::class, $news);
+        $news->setAuthor($security->getUser());
+        $form->handleRequest($request);
+        if($form->isSubmitted() && $form->isValid()){
+            $imageFile = $form->get('image')->getData();
+            if($imageFile){
+                $originalFilename = pathinfo($imageFile->getClientOriginalName(), PATHINFO_FILENAME);
+                $safeFilename = $slugger->slug($originalFilename);
+                $newFilename = $safeFilename.'-'.uniqid().'.'.$imageFile->guessExtension();
+
+                try {
+                    $imageFile->move(
+                        $this->getParameter('uploads_directory'),
+                        $newFilename
+                    );
+                } catch (FileException $e) {
+                    // Gérer l'exception en conséquence
+                }
+
+                $image = new Images();
+                $image->setFilename($newFilename);
+                $image->setName('Test');
+                $image->setAltText('Test encore');
+                $entityManager->persist($image);
```

Comment:
On peut s'eviter le persit si il y'a un lien entre une entité A et une entité B en precisant persist dans les relation Doctrine. Ne te prend pas la tete avec ça pour l'instant mais c'est pour info --> https://yusufbiberoglu.medium.com/cascade-persist-remove-in-doctrine-559ac967e451